### ISSUE TITLE
missing PAT in lock file

### ISSUE THREAD

**Issue by baixiaoyu2997** [pnpm/pnpm#7078]
### Verify latest release

- [X] I verified that the issue exists in the latest pnpm release

### Which area(s) of pnpm are affected? (leave empty if unsure)

Lockfile

### Reproduction steps

1. rm pnpm-lock.yml
2. pnpm i 
3. check lockfile

### Describe the Bug

In my project, I am referencing a private GitHub repository as a package. When my PNPM version is greater than 8.6.9, it causes the Personal Access Token (PAT) to be missing in the lockfile.

You can see that before the upgrade, there was a PAT in the version field, but after upgrading to 8.6.10, it is no longer present.
<img width="835" alt="image" src="https://github.com/pnpm/pnpm/assets/9149999/929465ae-d43b-42f5-8010-bebdf3bd0b55">

### Expected Behavior

have PAT in `version`

### Which Node.js version are you using?

18.16.0

### Which operating systems have you used?

- [X] macOS
- [ ] Windows
- [ ] Linux

**Comment by zkochan** [comment]
Probably this started happening after this was merged: https://github.com/pnpm/pnpm/pull/6806

cc @KSXGitHub

**Comment by KSXGitHub** [comment]
The token already presents in `specifier`. Why do you need it in `version` as well? Does it cause `pnpm install` to stop working?

**Comment by baixiaoyu2997** [comment]
> The token already presents in `specifier`. Why do you need it in `version` as well? Does it cause `pnpm install` to stop working?

can't work on vercel
<img width="1188" alt="image" src="https://github.com/pnpm/pnpm/assets/9149999/b7262ccf-8fe4-4a12-b51d-bc617e177433">

**Comment by KSXGitHub** [comment]
I see. So this bug can be fixed either by re-adding auth token to `version` or make pnpm fetch infer auth token from the specifier.

**Comment by KSXGitHub** [comment]
@baixiaoyu2997 In the meantime, you need to provide your vercel builder a private SSH key that has access to the private repositories.

**Comment by KSXGitHub** [comment]
@zkochan 
> Probably this started happening after this was merged: #6806 

This PR was later [reverted](https://github.com/pnpm/pnpm/issues/6805#issuecomment-1639992599) and replaced by https://github.com/pnpm/pnpm/pull/6832

**Comment by baixiaoyu2997** [comment]
> @baixiaoyu2997 In the meantime, you need to provide your vercel builder a private SSH key that has access to the private repositories.

yes, i have, the latest version still has this problem, so, i switched back to 8.6.9 again, any update on this?

**Comment by jon301** [comment]
Hi
I have the same issue on the latest version.
Pinning to 8.6.9 works fine

**Comment by ashbuilds** [comment]
> Hi I have the same issue on the latest version. Pinning to 8.6.9 works fine

You saved my another 24 hours that were about to go to waste, thanks!

**Comment by jon301** [comment]
Hi, do you have any news ?
Just tried with pnpm v10.0.0 and this issue still occur

**Comment by zkochan** [comment]
Closing: current pnpm covers this.

pnpm stops writing credentials that appear in a dependency URL into the lockfile, so a token is not committed with it. Provide the token through the git credential helper or an npmrc auth entry instead.

If you still hit this on the latest pnpm, please comment with a reproduction and we will reopen.

---
Written by an agent (Claude Code, claude-opus-5).

**Comment by zkochan** [comment]
Correction to my previous comment: I checked this on pnpm 12.5.1 and the credentials in a git dependency URL *are* written to the lockfile again.

With `"dep": "git+https://user:secret@github.com/owner/repo.git"` in package.json, the lockfile records:

```
  dep@git+https://user:secret@github.com/owner/repo.git#<commit>:
```

So what this issue reports no longer happens, and I am closing it as fixed rather than for the reason I gave before.

---
Written by an agent (Claude Code, claude-opus-5).
